analyzer-optimize with cpp.ConstCharStar generates invalid cpp

Sorry if this is the same root cause as #9732, it generates a different error and a different work around was needed so I decided to open a new issue.

When trying to compile some externs which used cpp.ConstCharStar with analyer-optimise compilation started failing at the cpp stage.

The below sample will fail with haxe 4.1.2 and hxcpp 4.1.15 when using analyer-optimise. 
```haxe
class Main {
	static function main() {
		someFunc('hello world');
	}

	static function someFunc(_str : cpp.ConstCharStar) {
		trace('test');
	}
}
```
MSVC gives the following error
```
Error: Main.cpp
./src/Main.cpp(33): error C2228: left of '.get_raw' must have class/struct/union
./src/Main.cpp(33): note: type is 'const char *'
```
With this being the generated cpp code
```cpp
void Main_obj::main(){
            	HX_STACKFRAME(&_hx_pos_e47a9afac0942eb9_6_main)
HXDLIN(   6)		::Main_obj::someFunc(( (const char*)( (HX_("hello world",c4,ee,7b,f8).utf8_str()).get_raw()) ));
            	}
```
I was able to work around this by modifying the constructor of cpp.ConstCharStar, instead of using untyped to access the `__s` property of the string either of the following two compile fine.
```haxe
inline function new(s:String)
		this = cpp.NativeString.raw(s);
inline function new(s:String)
		this = cpp.NativeString.c_str(s).constRaw;
```
They do generate different cpp code though, and I haven't checked if they both work the same when it comes to unicode strings.
```cpp
::Main_obj::someFunc(HX_("hello world",c4,ee,7b,f8).raw_ptr());
::Main_obj::someFunc(( (const char*)(cpp::Pointer_obj::fromPointer(HX_("hello world",c4,ee,7b,f8).c_str())->get_constRaw()) ));
```
Using a custom ConstCharStar abstract which uses NativeString.raw has been working fine as a workaround for my externs though.

--- a/src/generators/gencpp.ml
+++ b/src/generators/gencpp.ml
@@ -2636,7 +2636,7 @@ let retype_expression ctx request_type function_args function_type expression_tr
                else if fieldName="cca" && obj.cpptype=TCppString then
                   CppFunction( FuncInternal(obj,"cca","."), TCppScalar("int")), TCppDynamic
                else if fieldName="__s" && obj.cpptype=TCppString then
-                  CppVar( VarInternal(obj,".","utf8_str()")), TCppPointer("ConstPointer", TCppScalar("char"))
+                  CppVar( VarInternal(obj,".","utf8_str()")), TCppRawPointer("const ", TCppScalar("char"))
                else if fieldName="__Index" then
                   CppEnumIndex(obj), TCppScalar("int")
                else if is_internal_member fieldName || cpp_is_real_array obj then begin

